### PR TITLE
GEOPY-1910: Throw warning if any of the common auto-saving drive is found in mode="r+"

### PR DESCRIPTION
**Pull request by benk-mira**
**GEOPY-1910 - Throw warning if any of the common auto-saving drive is found in mode="r+"**

### CODE REVIEW

**Review by domfournier — DISMISSED** [review]

File: geoh5py/workspace/workspace.py [hunk 1]
```diff
@@ -1236,6 +1246,13 @@ def open(self, mode: str | None = None) -> Workspace:
         if mode is None:
             mode = self._mode
 
+        if mode == "r+" and any(k in str(self.h5file) for k in NETWORK_DRIVES):
```
Comment (domfournier) [inline geoh5py/workspace/workspace.py:1249]:
```suggestion
        if mode == "r+" and any(k in str(self.h5file.absolute()) for k in NETWORK_DRIVES):
```
Need to also make sure h5file is a `str` of `Path`.

File: geoh5py/workspace/workspace.py [hunk 2]
```diff
@@ -73,6 +73,16 @@
 )
 
 
+NETWORK_DRIVES = [
+    "Egnyte Connect",
+    "One Drive",
+    "Dropbox",
+    "Google Drive",
```
Comment (domfournier) [inline geoh5py/workspace/workspace.py:80]: Mine is `googledrive`

Comment (benk-mira) [reply geoh5py/workspace/workspace.py:80]: I think you must have updated it when installing, or they changed their default.  The current default is "Google Drive"

Comment (domfournier) [reply geoh5py/workspace/workspace.py:80]: It says Google Drive in the explorer, but the windows path is lower case. 
image

Anyway, doesn't cost much to add it to the list.

Comment (benk-mira) [reply geoh5py/workspace/workspace.py:80]: ok, added

**Review by domfournier — APPROVED** [review]

**Pull request merged**